Examples/Algorithms/Geant4/include/ActsExamples/Geant4/SensitiveSurfaceMapper.hpp

```diff
@@ -118,6 +122,20 @@ class SensitiveSurfaceMapper {
                            G4VPhysicalVolume* g4PhysicalVolume,
                            const Acts::Transform3& motherTransform) const;
 
+  /// Function that checks the success of the mapping, and exposes
+  /// some additional information for debugging
+  ///
+  /// @param state state object after a call to remapSensitiveNames
+  /// @param gctx the geometry context
+  /// @param writeMissingG4VolsAsObj write the Geant4 volumes that are
+  /// not mapped to an obj file
+  /// @param writeMissingSurfacesAsObj write the sensitive surfaces that
+  /// where not mapped to an obj file
+  /// @return Returns true only if all sensitive surfaces where mapped
+  bool checkMapping(const State& state, const Acts::GeometryContext& gctx,
+                    bool writeMissingG4VolsAsObj = false,
+                    bool writeMissingSurfacesAsObj = false) const;
```

Comment:
Just looking at the interface I would wonder where these files are written and why I don't have the control over that. I guess this is mostly a debugging utility so maybe we should hide the parameters from the outside and have just the flags and pathnames inside the function?

Reply:
Hmm so the check mapping function is now also called within the Simulation now, so I need some flags to controll the writing of the files...